q2_feature_table/plugin_setup.py

```diff
@@ -236,6 +236,41 @@
                 "the data from the first will be propagated to the result."
 )
 
+plugin.methods.register_function(
+    function=q2_feature_table.rename_samples,
+    inputs={
+        'table': FeatureTable[Frequency | RelativeFrequency],
+    },
+    parameters={
+        'metadata': MetadataColumn[Categorical],
+        'strict': Bool,
+        'axis': Str % Choices({'sample', 'feature'})
+        },
+    outputs=[
+        ('renamed_table', FeatureTable[Frequency | RelativeFrequency])
```

Comment:
I like where you're going with this, but we'll need to use a `TypeMatch` in order to make this work:

```python
T = TypeMatch([Frequency, RelativeFrequency])

plugin.methods.register_function(
    ...
    inputs={'table': FeatureTable[T]},
    ...
    outputs={'renamed_table': FeatureTable[T]},
    ...
)
```

This tells QIIME 2 to return a `FeatureTable[Frequency]` when it receives a `FeatureTable[Frequency]` as input, and similarly with `RelativeFrequency`.

You could also expand that `TypeMatch` to include all kinds of other feature tables subtypes

```python
TypeMatch([Frequency, RelativeFrequency, PresenceAbsence, Composition])
```

Reply:
I didn't know this was possible. super cool!